include/hipSYCL/sycl/libkernel/generic/hiplike/builtins.hpp

```diff
@@ -396,6 +396,19 @@ HIPSYCL_HIPLIKE_BUILTIN T __hipsycl_clamp(T x, T minval, T maxval) noexcept {
     hiplike_builtins::__hipsycl_max(x, minval), maxval);
 }
 
+template<class T>
+HIPSYCL_HIPLIKE_BUILTIN T __hipsycl_clz(T x) noexcept {
+
+  // use __clzll or __clz by checking the bit lenght because
+  // the nvidia/hip documentation mention clz as 32 bits and clzll as 64
+  
+  if constexpr (sizeof(T)*CHAR_BIT == 64){
```

Comment:
`sizeof(T) == 8`?

Reply:
Technically the standard doesn't enforce a char to be 8 bits but this one is guaranteed to be correct `sizeof(T)*CHAR_BIT`. I don't know if that's the case for any architectures that may be of use tho.

Reply:
That's technically true, but I don't think any architecture that we target has non-8 bit char width. I think we also assume `CHAR_BIT==8` at other places. And likely heterogeneous compute will fall completely apart if the device does not have `CHAR_BIT` of 8 :-)
It's okay for me if you prefer it this way, just want to let you know that assuming `CHAR_BIT==8` is a legitimate option if it can simplify code :-)

Reply:
At some point, if you want to run for example on a Zuse 3 or its emulator, you might have 22-bit characters for example. ;-)
```suggestion
  if constexpr (sizeof(T)*CHAR_BIT > 32) {
```
to handle an Honeywell GCOS 6 with 36 bit integers. ;-)